main.go

```diff
@@ -49,13 +47,45 @@ func main() {
 	rewrites = make(map[string]string)
 	visited = make(map[string]bool)
 
-	err := vendorize(pkgName, dest)
+	err := vendorize(pkgName, chooseGOPATH(gopaths, dest), dest)
 	if err != nil {
 		log.Fatal(err)
 	}
 }
 
-func vendorize(path, dest string) error {
+// chooseGOPATH selects the gopath component that has the longest prefix in common with dest.
+// It breaks ties by preferring earlier components.
+func chooseGOPATH(gopaths []string, dest string) string {
+	for {
+		dest = strings.TrimSuffix(dest, "/")
+		chosen := ""
+		for i := len(gopaths) - 1; i >= 0; i-- {
+			dir := filepath.Join(gopaths[i], "src", dest)
+			if _, err := os.Stat(dir); err == nil {
+				chosen = gopaths[i]
+			}
+		}
+		if chosen != "" {
+			return chosen
+		}
+
+		dest, _ = filepath.Split(dest)
+		if dest == "" {
+			break
+		}
+	}
+
+	// None of the gopaths contain any prefix of dest.
```

Comment:
Hm, I guess we are preferring earlier GOPATH entries rather than later, so what you've done makes sense then...